pylint/checkers/typecheck.py

```diff
@@ -1720,9 +1720,18 @@ def visit_subscript(self, node):
             return
 
         inferred = safe_infer(node.value)
+
         if inferred is None or inferred is astroid.Uninferable:
             return
 
+        if inferred.decorators:
+            first_decorator = helpers.safe_infer(inferred.decorators.nodes[0])
+            if isinstance(first_decorator, astroid.ClassDef):
+                inferred = first_decorator.instantiate_class()
+            else:
+                return  # It would be better to handle function
+                # decorators, but let's start slow.
```

Comment:
No need to handle this for now. We are not sure it will be useful one day and it become the case it will be easy to add lines here.

Reply:
That's why I added a comment, to guide future implementers it's here that it should be implemented. But I can reword this more lightly like "If we need to handle function decorators, do it here"? Or I can drop the comment if you prefer.

Reply:
Yes it could be reworded. Maybe something along the lines: `If necessary handling function decorators should be here`